pkg/core/runtime/component/resilient.go

```diff
@@ -42,23 +42,38 @@ func (r *resilientComponent) Start(stop <-chan struct{}) error {
 					}
 				}
 			}()
-
 			errCh <- r.component.Start(stop)
 		}(errCh)
-		select {
-		case <-stop:
-			r.log.Info("done")
-			return nil
-		case err := <-errCh:
-			if err != nil {
-				r.log.WithValues("generationID", generationID).Error(err, "component terminated with an error")
+
+		receivedErr := false
+		for !receivedErr {
+			select {
+			case <-stop:
+				r.log.Info("done")
+				return nil
+			case err := <-errCh:
+				if err != nil {
+					r.log.WithValues("generationID", generationID).Error(err, "component terminated with an error")
+				}
+				receivedErr = true
+			case <-time.After(backoffMaxTime):
+				// reset backoff so in a case of events
+				// 1) Component is unhealthy until max backoff is reached
+				// 2) Component is healthy for at least backoff max time
+				// 3) Component is unhealthy
```

Comment:
I don't understand 1 and 3 - IMO we should only reset backoff when the component is healthy again. Also isn't 3 just superset of 1 so we don't need 1?

Reply:
it's steps. First step then second step then third